### ISSUE TITLE
Integration stopped working with HA OS 13.0 - webhook could not be deleted

### ISSUE THREAD

**Issue by Plawasan** [craibo/ha_strava#178]
I've just updated to HA OS 13.0 and since the restart I get no entities from the Strava integration with these errors in logs:

```
This error originated from a custom integration.

Logger: custom_components.ha_strava
Source: custom_components/ha_strava/__init__.py:920
integration: Strava (Unofficial) (documentation, issues)
First occurred: 14:41:03 (2 occurrences)
Last logged: 14:43:36

Strava webhook for https://<myid>.ui.nabu.casa/api/strava/webhook could not be deleted: {"message":"Resource Not Found","errors":[{"resource":"PushSubscription","field":"","code":"not found"}]}
```

```
This error originated from a custom integration.

Logger: custom_components.ha_strava
Source: custom_components/ha_strava/__init__.py:887
integration: Strava (Unofficial) (documentation, issues)
First occurred: 14:43:35 (1 occurrences)
Last logged: 14:43:35

Bad logger message: Strava failed to remove listener ((ValueError('list.remove(x): x not in list'),))
```

It could of course be just a coincidence with the OS upgrade however that's the last update I've made to my system which previously worked OK.

Let me know if you need any other info to check this.

Thanks

**Comment by Plawasan** [comment]
A day without any changes/restarts on my side the integration is working again. I'll leave the issue open for now in case someone else hits the same problem, will close it later if not.

**Comment by DarwinData** [comment]
Exactly same problem as OP for me.

**Comment by craibo** [comment]
Hey,

I have been trying to replicate this and the only way I can do it, is if the ha callback url or SSL/TLS fails (invalid cert, url not working, etc.) 

Restarting the integration, if your public url is working, will resolve the issue. Alternatively the integration will check daily for new data and as part of that will re-attempt to register the callback if it detects its not setup.

Closing this out.

**Comment by craibo** [comment]
I have managed to replicate this and can address the issue.

**Comment by craibo** [comment]
This is resolved with release 3.2.42
